staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```diff
@@ -156,6 +170,11 @@ func (a customResourceStrategy) Validate(ctx context.Context, obj runtime.Object
 
 		// validate x-kubernetes-list-type "map" and "set" invariant
 		errs = append(errs, structurallisttype.ValidateListSetsAndMaps(nil, a.structuralSchemas[v], u.Object)...)
+
+		// validate x-kubernetes-validations rules
+		if a.celValidators != nil {
+			errs = append(errs, a.celValidators[v].Validate(nil, a.structuralSchemas[v], u.Object)...)
```

Comment:
can `a.celValidators[v]` be nil here if there were no `x-kubernetes-validators` fields in the schema? (if not, should it be nil?)

would this be clearer:
```go
if celValidator, ok := a.celValidators[v]; ok {
  errs = append(errs, celValidator.Validate(nil, a.structuralSchemas[v], u.Object)...)
}
```

same questions below in `customResourceStrategy#ValidateUpdate` and in `statusStrategy#ValidateUpdate`

Reply:
Yes, this is what I ended up with. It worked out pretty clean.